api/v1alpha1/util.go

```diff
@@ -12,18 +11,19 @@ const (
 
 // Type returns the type of the action.
 func (t HookAction) Type() (string, error) {
-	var data map[string]interface{}
-	if err := json.Unmarshal(t.union, &data); err != nil {
-		return "", err
+	// The OpenAPI spec will prevent this from happening, but we should still
+	// evaluate for go code usage.
```

Comment:
Are you sure that the OpenAPI spec will prevent this?  I'm not familiar with the syntax you used and don't see the check in the generated code.

Reply:
yeah I am reworking this